### DISCUSSION TITLE
How to use Controller with MUI Autocomplete in Typescript

### DISCUSSION THREAD

**Discussion by ronaldruzicka** [react-hook-form/react-hook-form#2484] [Q&A]
Hi guys, I'm struggling to use MUI Autocomplete with Controller in Typescript.

I have a `Select` component that is wrapped in a `Controller` which renders `Autocomplete` component from Material UI. Since `control` is taking a generic `T` for it's `Control<T>` type, it's inferring a whole `FormValues` type. The `Autocomplete` is also taking `T` for it's `options` prop. But then I get typescript errors, because `FormValues` is not assignable to `CountryType`. Also I need the `getOptionLabel` and `getOptionSelected` as props for my select, because I need to pass custom `options`(`string[]` or custom `object[]`) and I don't have the correct types there.

Do you have any idea how to have the correct types? I don't want to use `any`, because I want to see what the actual `option` type is.

Here is a code sandbox: https://codesandbox.io/s/material-demo-r730l?file=/Select.tsx

Thanks! 🙏🏻

**Comment by bluebill1049** [comment]
are you using `render` prop?

**Reply by ronaldruzicka** [reply]
yes, it's in the `Select.tsx` component in the code sandbox. The error is in the `demo.tsx`.

**Reply by bluebill1049** [reply]
maybe this helps? https://codesandbox.io/s/react-hook-form-v6-controller-ts-4dpm9?file=/src/MuiAutoComplete.tsx

**Reply by ronaldruzicka** [reply]
Thanks, I'll check it out 🙂
